components/org.wso2.carbon.identity.tenant.resource.manager/src/main/java/org/wso2/carbon/identity/tenant/resource/manager/TenantAwareAxis2ConfigurationContextObserver.java

```diff
@@ -0,0 +1,188 @@
+
+        List<EventStreamConfiguration> eventStreamConfigurationList = null;
+        try {
+            eventStreamConfigurationList = TenantResourceManagerDataHolder.getInstance().getCarbonEventStreamService()
+                    .getAllEventStreamConfigurations();
+        } catch (EventStreamConfigurationException e) {
+            log.error(handleServerException(
+                    TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_FETCHING_SUPER_TENANT_EVENT_STREAM_CONFIGURATION,
+                    e, String.valueOf(tenantId)));
+        } finally {
+            PrivilegedCarbonContext.endTenantFlow();
+        }
+        return eventStreamConfigurationList;
+    }
+
+    private void startTenantFlow(int tenantId, String tenantDomain) {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
+    }
+
+    private void startSuperTenantFlow() {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        carbonContext.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
+        carbonContext.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+    }
+
+    private void createTenantEventPublisherConfiguration(int tenantId,
+            List<EventPublisherConfiguration> activeEventPublisherConfigurations) {
+
+        for (EventPublisherConfiguration eventPublisherConfiguration : activeEventPublisherConfigurations) {
+            try {
+                ResourceFile resourceFile = getResourceFile(eventPublisherConfiguration.getEventPublisherName());
+                if (resourceFile != null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Configuration store is enabled and there is a file in the configuration store for "
+                                + "the publisher name: " + eventPublisherConfiguration.getEventPublisherName());
+                    }
+                    getResourceManager().addEventPublisherConfiguration(resourceFile);
+                }
+            } catch (ConfigurationManagementException e) {
+                if (e.getErrorCode()
+                        .equals(ConfigurationConstants.ErrorMessages.ERROR_CODE_FEATURE_NOT_ENABLED.getCode())) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Configuration store is disabled.");
+                    }
+                }
+            } catch (TenantResourceManagementException e) {
+                log.error(handleServerException(
+                        ERROR_CODE_ERROR_WHEN_CREATING_TENANT_EVENT_PUBLISHER_CONFIGURATION_BY_CONFIG_STORE, e,
+                        eventPublisherConfiguration.getEventPublisherName(), String.valueOf(tenantId)));
```

Comment:
Rather than using tenant id better to use tenant domain in logs. So that it will be more readable

Reply:
done